File: Typescript2/6.R
struct ArrayWrapper {
    array: Vec<i32>,
}

impl ArrayWrapper {
    fn new(array: Vec<i32>) -> Self {
        Self { array }
    }

    fn add_arrays(arr1: &ArrayWrapper, arr2: &ArrayWrapper) -> ArrayWrapper {
        let mut sum_array = Vec::new();
        let max_length = std::cmp::max(arr1.array.len(), arr2.array.len());

        for i in 0..max_length {
            let sum = arr1.array.get(i).unwrap_or(&0) + arr2.array.get(i).unwrap_or(&0);
            sum_array.push(sum);
        }

        ArrayWrapper::new(sum_array)
    }

    fn to_string(&self) -> String {
        let elements: Vec<String> = self.array.iter().map(|x| x.to_string()).collect();
        format!("[{}]", elements.join(","))
    }
}

fn main() {
    let arr1 = ArrayWrapper::new(vec![1, 2, 3]);
    let arr2 = ArrayWrapper::new(vec![4, 5, 6]);

    let sum_array = ArrayWrapper::add_arrays(&arr1, &arr2);
    println!("{}", sum_array.to_string()); // Output: [5,7,9]
}

File: Typescript2/6.cpp
#include <vector>
#include <algorithm>

class ArrayWrapper {
private:
    std::vector<int> array;

public:
    ArrayWrapper(std::vector<int> array) : array(array) {}

    static ArrayWrapper addArrays(const ArrayWrapper& arr1, const ArrayWrapper& arr2) {
        std::vector<int> sumArray;
        int maxLength = std::max(arr1.array.size(), arr2.array.size());
        sumArray.reserve(maxLength);

        for (size_t i = 0; i < maxLength; ++i) {
            int sum = (i < arr1.array.size() ? arr1.array[i] : 0) + (i < arr2.array.size() ? arr2.array[i] : 0);
            sumArray.push_back(sum);
        }

        return ArrayWrapper(sumArray);
    }

    std::string toString() const {
        std::string result = "[";
        for (size_t i = 0; i < array.size(); ++i) {
            result += std::to_string(array[i]);
            if (i < array.size() - 1) {
                result += ",";
            }
        }
        result += "]";
        return result;
    }
};

int main() {
    ArrayWrapper arr1({1, 2, 3});
    ArrayWrapper arr2({4, 5, 6});

    ArrayWrapper sumArray = ArrayWrapper::addArrays(arr1, arr2);
    std::cout << sumArray.toString() << std::endl; // Output: [5,7,9]

    return 0;
}
